code/game/objects/structures/kitchen_spike.dm

```diff
@@ -112,16 +114,76 @@
 				user << "<span class='danger'>The spike already has something on it; finish collecting its meat first!</span>"
 			return
 		if(istype(G.affecting, /mob/living/carbon/human))
-			user.changeNext_move(CLICK_CD_MELEE)
-			playsound(src.loc, "sound/effects/splat.ogg", 25, 1)
-			var/mob/living/carbon/human/H = G.affecting
-			H.visible_message("<span class='danger'>[user] slams [G.affecting] into the meat spike!</span>", "<span class='userdanger'>[user] slams you into the meat spike!</span>", "<span class='italics'>You hear a squishy wet noise.</span>")
-			H.adjustBruteLoss(20)
-			return
+			if(!occupied)
+				if(do_mob(user, src, 50))
+					if(occupied) //to prevent spam/queing up attacks
+						return
+					if(G.affecting.buckled)
+						return
+					var/mob/living/carbon/human/H = G.affecting
+					playsound(src.loc, "sound/effects/splat.ogg", 25, 1)
+					H.visible_message("<span class='danger'>[user] slams [G.affecting] onto the meat spike!</span>", "<span class='userdanger'>[user] slams you onto the meat spike!</span>", "<span class='italics'>You hear a squishy wet noise.</span>")
+					H.loc = src.loc
+					H.emote("scream")
+					var/turf/pos = get_turf(H)
+					pos.add_blood_floor(H)
+					H.adjustBruteLoss(30)
+					var/matrix/m120 = matrix(transform)

```

Comment:
This doesn't need the ```transform``` argument here, technically you're making a matrix of the MEAT SPIKE's transformation, and applying it to the human. Normally the meatspike's transform will just be an identity matrix, but you can never be sure with badmins

```matrix(transform)``` --> ```matrix()```